tests/unit/serve/executors/test_executor.py

```diff
@@ -331,3 +337,22 @@ def test_serve(served_exec):
     docs = Client(port=PORT).post(on='/foo', inputs=DocumentArray.empty(5))
 
     assert docs.texts == ['foo' for _ in docs]
+
+
+def test_set_workspace():
```

Comment:
use `tmpdir` fixture to make sure test uses a temporary directory and does not pollute user system (ours)